File: php/features.php
<?php
declare(strict_types=1);

// Elonmuskapi SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ElonmuskapiFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ElonmuskapiBaseFeature();
            case "test":
                return new ElonmuskapiTestFeature();
            default:
                return new ElonmuskapiBaseFeature();
        }
    }
}

File: py/elonmuskapi_sdk/features.py
# Elonmuskapi SDK feature factory

from elonmuskapi_sdk.feature.base_feature import ElonmuskapiBaseFeature
from elonmuskapi_sdk.feature.test_feature import ElonmuskapiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ElonmuskapiBaseFeature(),
        "test": lambda: ElonmuskapiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
